src/main/js/diff/Diff.tsx

```diff
@@ -64,99 +65,81 @@ const CommentWrapper = styled.div`
-  // path
-  [key: string]: FileCommentState;
-};
-export type LineCommentCollection = {
-  // hunkid
-  [key: string]: {
-    // changeid
-    [key: string]: {
-      location: Location;
-      comments: string[];
-      editor?: boolean;
-    };
-  };
-};
-
-export type DiffState = {
-  files: FileCommentCollection;
-  lines: LineCommentCollection;
-  comments: Comment[];
-  reviewedFiles: string[];
-};
-
 type Props = {
   repository: Repository;
   pullRequest: PullRequest;
+  comments?: Comments;
   diffUrl: string;
-  diffState: DiffState;
-  updateDiffState: (state: DiffState) => void;
   createLink?: string;
   fileContentFactory: FileContentFactory;
-};
-
-const openEditor = (
-  diffState: DiffState,
-  updateDiffState: (state: DiffState) => void,
-  editor: boolean,
-  location?: Location
-) => {
-  if (isInlineLocation(location)) {
-    const lineComments = diffState.lines;
-    const hunkId = createHunkIdFromLocation(location!);
-    const changeId = createChangeIdFromLocation(location!);
-    const hunkComments = lineComments[hunkId] || {};
-    const changeComments = hunkComments[changeId] || {
-      location,
-      comments: []
-    };
-
-    changeComments.editor = editor;
-    hunkComments[changeId] = changeComments;
-    lineComments[hunkId] = hunkComments;
-    updateDiffState({ ...diffState, lines: lineComments });
-  } else {
-    const fileComments = diffState.files;
-    const file = location!.file;
-    const fileState = fileComments[file] || {
-      comments: []
-    };
-    fileState.editor = editor;
-    fileComments[file] = fileState;
-    updateDiffState({ ...diffState, files: fileComments });
-  }
+  reviewedFiles: string[];
 };
 
 const Diff: FC<Props> = ({
   repository,
   pullRequest,
+  comments,
   diffUrl,
-  diffState,
-  updateDiffState,
   createLink,
-  fileContentFactory
+  fileContentFactory,
+  reviewedFiles
 }) => {
   const [t] = useTranslation("plugins");
   const { actions, isCollapsed } = useDiffCollapseState(pullRequest);
   const [collapsed, setCollapsed] = useState(false);
+  const [openEditors, setOpenEditors] = useState<{ [hunkId: string]: string[] }>({});
+
+  const openEditor = (editor: boolean, location: Location) => {
+    if (isInlineLocation(location)) {
+      if (editor) {
+        const preUpdateOpenEditors = openEditors[createHunkIdFromLocation(location)] || [];
+        setOpenEditors(prevState => ({
+          ...prevState,
+          [createHunkIdFromLocation(location)]: [
+            ...preUpdateOpenEditors,
+            !preUpdateOpenEditors.includes(createChangeIdFromLocation(location))
+              ? createChangeIdFromLocation(location)
+              : ""
+          ]
+        }));
+      } else {
+        setOpenEditors(prevState => ({
+          ...prevState,
+          [createHunkIdFromLocation(location)]: [
+            ...prevState[createHunkIdFromLocation(location)].filter(l => l !== createChangeIdFromLocation(location))
+          ]
+        }));
```

Comment:
Doesn't really matter here, but computing `createHunkIdFromLocation` and `createChangeIdFromLocation` only once would be nice and maybe even make the code more readable.
```suggestion
        const hunkId = createHunkIdFromLocation(location);
        const changeId = createChangeIdFromLocation(location);
        setOpenEditors(prevState => ({
          ...prevState,
          [hunkId]: [
            ...prevState[hunkId].filter(l => l !== changeId)
          ]
        }));
```